finagle-core/src/test/scala/com/twitter/finagle/builder/EndToEndTest.scala

```diff
@@ -196,5 +208,9 @@ class EndToEndTest extends FunSuite {
 
     assert(requests === 1)
     assert(triesRequests === 1)
+
+    // NOTE by timxzl: need to properly close the client
+    // otherwise it will prevent ExitGuard from exiting and interfere with ExitGuardTest
+    Await.ready(client.close(), 1.second)
```

Comment:
since we need to do this so many places, is there a way to help get it done generically?